x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/QueryTranslatorTests.java

```diff
@@ -2459,4 +2461,33 @@ public void testReplaceSumWithStats() {
             assertThat(eqe.queryContainer().toString().replaceAll("\\s+", ""), containsString("{\"stats\":{\"field\":\"int\"}}"));
         }
     }
+
+    public void testAddMissingEqualsToBoolField() {
+        LogicalPlan p = plan("SELECT bool FROM test WHERE bool");
```

Comment:
Maybe I'm missing some context, but what happens to nested expressions like: `WHERE a > 1 and (bool or b < 2) or (c = 3 and bool)`, could we have a test for that as well?

Reply:
They'd be eval'd to a `BOOLEAN` type, so nothing would happen: the `AddMissingEqualsToBoolField()` wouldn't change the expression (since its root node type wouldn't be a FieldAttribute) and the verifier check would let it pass (since it's a bool type). Basically most of the tests we have look to these two newly added checks similar to your example, but I could still add one if you think it'd help.

Reply:
@matriv, I've added a slightly extended version of your test (and updated my first eval of the case above).